### ISSUE TITLE
K8s upgrade to 1.22.1 - Failed to create Ingress 'XXX' because: the server could not find the requested resource (post ingresses.extensions)

### ISSUE THREAD

**Issue by vgn-tl** [hashicorp/terraform-provider-kubernetes#1386]
### Terraform Version, Provider Version and Kubernetes Version
```
Terraform version: 1.0.5
Kubernetes provider version: 2.4.1
Kubernetes version: 1.22.1
```

### Affected Resource(s)
- kubernetes-ingress

### Terraform Configuration Files
```hcl
# main.tf
resource "kubernetes_ingress" "example_ingress" {
  metadata {
    name = "example-ingress"
  }

  spec {
    backend {
      service_name = "MyApp1"
      service_port = 8080
    }

    rule {
      http {
        path {
          backend {
            service_name = "MyApp1"
            service_port = 8080
          }

          path = "/app1/*"
        }

        path {
          backend {
            service_name = "MyApp2"
            service_port = 8080
          }

          path = "/app2/*"
        }
      }
    }

    tls {
      secret_name = "tls-secret"
    }
  }
}
# versions.tf
terraform {
  required_providers {
    kubernetes = {
      source = "hashicorp/kubernetes"
      version = ">= 2.4.1"
    }
  }
  required_version = ">= 1.0.5"
}
# provider.tf
provider "kubernetes" {
  host = XXX
  client_certificate = XXX
  client_key = XXX
  cluster_ca_certificate = XXX
}
```

### Panic Output
```
kubernetes_ingress.example_ingress: Creating...
╷
│ Error: Failed to create Ingress 'default/example-ingress' because: the server could not find the requested resource (post ingresses.extensions)
│
│   with kubernetes_ingress.example_ingress,
│   on main.tf line 1, in resource "kubernetes_ingress" "example_ingress":
│    1: resource "kubernetes_ingress" "example_ingress" {
```

### Steps to Reproduce
1. `terraform apply`

### Expected Behavior
The run should have completed succesfully.

### Actual Behavior
After upgrading k8s to v1.22.1 and running a plan with a kubernetes_ingress controller, the ingress could not be created because the server could not find the requested resource (post ingresses.extensions)

### Important Factoids
We are using Exoscale SKS as kubernetes service and terraform.io cloud. 
We have tested the above configuration locally as well.

**Comment by vgn-tl** [comment]
Hi @hashicorp,
any news or update regarding this topic?

We haven't noticed any changes in the main branch yet:

https://github.com/hashicorp/terraform-provider-kubernetes/blob/28e968e0fc08e39d5a3af50015a4df7c4c58de83/kubernetes/resource_kubernetes_ingress.go#L177

Thank you,
kind regards

**Comment by dsonck92** [comment]
I'm currently attempting to write a PR for this since I'm blocked by this issue. Also since I cannot work around it with the new `kubernetes_manifest` (And since 1.22 is pretty close, and the old Ingress API has been deprecated for a long time, this should be updated anyways)

**Comment by eake** [comment]
Came across same problem, we had to downgrade kube cluster. Ive tried to update the provider, it works now with 1.22.1 (tested for 3 days). 
Im building the provider in our terraform image:
```
COPY ./terraform-provider-kubernetes /code
RUN apk --no-cache add jq git go && cd /code && go install && chmod 755 /root/go/bin/terraform-provider-kubernetes
```
And after that I use terraform.rc to use this provider:
```
provider_installation {
  dev_overrides {
    "hashicorp/kubernetes" = "/Users/wilhelmschmidtmayer/golang-bin/"
  }
  direct {}
}
```
And use it via tf var:
```
export TF_CLI_CONFIG_FILE=/path/to/terraform.rc
```

And it works against the 
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:39:34Z", GoVersion:"go1.16.7", Compiler:"gc", Platform:"linux/amd64"}

heres the fork: https://github.com/eake/terraform-provider-kubernetes

**Comment by dsonck92** [comment]
@eake I'm curious how well creating a new ingress works, since `pathType` is required to be set and I did not see anything related to it. Besides now not allowing named ports to be used (port.name). Have you ran the full test suite?

**Comment by mab-wien** [comment]
@alexsomesan

**Comment by vgn-mab** [comment]
@alexsomesan

**Comment by nickreynke** [comment]
@alexsomesan Sorry for pinging again but this is urgent. As @twittyc mentioned: There are some PRs ready to address this issue. When will we see any updates regarding this? We are blocked and can not create an Ingress, unless this gets resolved.

**Comment by vgn-tl** [comment]
@nickreynke while we wait for a new release, we have built a custom release and use this in terraform cloud as described here: [https://www.terraform.io/docs/cloud/run/install-software.html#in-house-providers](https://www.terraform.io/docs/cloud/run/install-software.html#in-house-providers)

**Comment by anuj-kosambi** [comment]
@vgn-tl  Thanks
placing custom build binary at `~/.terraform.d/pluginsregistry.terraform.io/hashicorp/kubernetes/2.6.2/linux_amd64/terraform-provider-kubernetes`
and replacing `resource "kubernetes_ingress" "name {` with `resource "kubernetes_ingress_v1" "name" {` supports v1.22 very well.

**Comment by chrisroat** [comment]
Are the necessary commits in to support this?  If yes, is there a rough expectation on the next release (days vs weeks)?

**Comment by chrisroat** [comment]
The answer to my question: days!  Version 2.7.0 of the provider supports 1.22 for ingress.

For me, I had to make two changes for my simple ingress without any rules.


First, change to using the new v1:

```
resource "kubernetes_ingress" "frontend" {
```

```
resource "kubernetes_ingress_v1" "frontend" {
```

Second, update how the spec was formed:

```
  spec {
    backend {
      service_name = kubernetes_service.frontend.metadata.0.name
      service_port = kubernetes_service.frontend.spec.0.port.0.port
    }
  }
```

```
 spec {
    default_backend {
      service {
        name = kubernetes_service.frontend.metadata.0.name
        port {
          number = kubernetes_service.frontend.spec.0.port.0.port
        }
      }
    }
  }
```

**Comment by lsascha** [comment]
Unfortunately, i wasn't able to use the custom binary using gitlabs terraform image.
`gitlab-terraform init` does work. But `gitlab-terraform validate` directly after, does fail with
```
Error: Could not load plugin
│ 
│ 
│ Plugin reinitialization required. Please run "terraform init".
```

Could be a gitlab-terraform image issue, but still annoying. Really thinking to get away from terraform if updates will continue to take that long or longer in the future. :disappointed:

**Comment by vgn-tl** [comment]
Yes, version 2.7.0 has finally been released as a tag (no github release yet, also available here https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs)

Updating to kubernetes_ingress_v1 works good in a green field environment.
Updating an already existing ingress can be a struggle, as terraform has trouble creating&deleting an ingress with the same name, even though it notices the difference between kubernetes_ingress vs kubernetes_ingress_v1.

We ended up removing the existing beta ingresses, after that another run created the new ingresses just fine.

**Comment by MFAshby** [comment]
This is still a problem on version 2.7.1, using latest microk8s
```
$ snap list microk8s
Name      Version  Rev   Tracking     Publisher   Notes
microk8s  v1.22.4  2695  1.22/stable  canonical✓  classic
```

From a fresh install of microk8s, enable `ingress` and `dns` addon and try to apply a terraform file like the original example, but ensure that kubernetes provider is upgraded to 2.7.1. It fails with the same error.

**Comment by vgn-tl** [comment]
resolved, see release 2.7.0 (https://github.com/hashicorp/terraform-provider-kubernetes/releases/tag/v2.7.0)

Thank you @jrhouston @alexsomesan !
